### ISSUE TITLE
native-lib-loader-2.3.0: NullPointerException in NativeLibraryUtil.java 

### ISSUE THREAD

**Issue by bimoelle** [scijava/native-lib-loader#23]
We depend on pom-scijava and use parts of it indirectly via Loci to read images from file. After updating to pom-scijava 23.1.0 some of our unit tests reading images from file crash with the following error:

```
java.lang.NullPointerException 
	at org.scijava.nativelib.NativeLibraryUtil.getPlatformLibraryPath(NativeLibraryUtil.java:189)
	at org.scijava.nativelib.NativeLibraryUtil.loadNativeLibrary(NativeLibraryUtil.java:308)  
	at loci.formats.services.JPEGTurboServiceImpl.<init>(JPEGTurboServiceImpl.java:107) 
	at loci.formats.in.NDPIReader.<init>(NDPIReader.java:69)  
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at loci.formats.ImageReader.<init>(ImageReader.java:129)
	at loci.formats.in.FilePatternReader.<init>(FilePatternReader.java:77)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at loci.formats.ImageReader.<init>(ImageReader.java:129)
	at loci.formats.ImageReader.<init>(ImageReader.java:116)
	at de.unihalle.informatik.MiToBo.io.images.ImageReaderMTB.initReader(ImageReaderMTB.java:341)
	at de.unihalle.informatik.MiToBo.io.images.ImageReaderMTB.setFileName(ImageReaderMTB.java:866)
	at de.unihalle.informatik.MiToBo.io.images.ImageReaderMTB.<init>(ImageReaderMTB.java:206)
	...
```

Tracing back the problem it turned out that pom-scijava 23.0.0 worked without problems. The update from native-lib-loader-2.2.0 to native-lib-loader-2.3.0 which comes with pom-scijava 23.1.0 seems to cause the problems. It seems that commit b0e0fd975ccbbae890a2b326135f8e59bde7612e introduced an error in the method getPlatformLibraryPath(...) in [org/scijava/nativelib/NativeLibraryUtil.java](https://github.com/scijava/native-lib-loader/blob/master/src/main/java/org/scijava/nativelib/NativeLibraryUtil.java). This method is called in line 308 with `null` as argument, however, in line 189 the `null` argument is accessed without further checks. As the stack trace above shows this seems to be the source of the null pointer exception.

**Comment by imagejan** [comment]
Likely fixed by #22, once it gets merged.

See also [this forum post](https://forum.image.sc/t/fiji-does-not-open-deltavision-dv-files-with-bioformat-importer-window-anymore/11892/2?u=imagejan).

**Comment by bimoelle** [comment]
Thanks for the quick reply and for pointing me towards the forum post and the pull request. I searched for corresponding posts and bug reports before, but obviously I overlooked all of them - sorry for that.

**Comment by ctrueden** [comment]
Sorry for the delay in getting this fix deployed. I merged #22, so I'll close this issue. I'm going to do some quick tests and then cut a new release of native-lib-loader today. If there continue to be problems, please reopen this issue and/or open new issues accordingly.
